### DISCUSSION TITLE
1.31.0

### DISCUSSION THREAD

**Discussion by dani-garcia** [dani-garcia/vaultwarden#4716] [General]
## Major changes and New Features
* Initial support for the beta releases of the new native mobile apps
* Removed support for WebSocket traffic on port 3012, as it's been integrated on the main HTTP port for a few releases
* Updated included web vault to 2024.5.1

## General mention
Bitwarden has changed the push API endpoints which affects the EU region endpoint users.
So if you use the push functionality and use the EU region you need to make some changes.
You have to update `push.bitwarden.eu` to `api.bitwarden.eu`.
This is also an issue with any previous version of Vaultwarden.

## What's Changed
* chore: remove repetitive words by @one230six in https://github.com/dani-garcia/vaultwarden/pull/4422
* Fix comment in events.rs by @KrappRamiro in https://github.com/dani-garcia/vaultwarden/pull/4408
* Improve JWT RSA key initialization and avoid saving public key by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4085
* Remove custom WebSocket code by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4001
* refactor: replace panic with a graceful exit by @tessus in https://github.com/dani-garcia/vaultwarden/pull/4402
* Small improvements around email change by @Timshel in https://github.com/dani-garcia/vaultwarden/pull/4415
* Change timestamp data type. by @gzfrozen in https://github.com/dani-garcia/vaultwarden/pull/4355
* Fix #3624: fix manager permission within groups by @matlink in https://github.com/dani-garcia/vaultwarden/pull/3754
* automatically use email address as 2fa provider by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4317
* fix: typos by @testwill in https://github.com/dani-garcia/vaultwarden/pull/4440
* Update chrono and sqlite by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4436
* Update Rust and crates by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4445
* Use async verify for Yubikey by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4448
* update web-vault to v2024.3.1 (new vertical layout) by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4468
* Update crates and some Clippy fixes by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4475
* Update Key Rotation web-vault v2024.3.x by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4446
* Update Crate and Rust by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4522
* Implement custom DNS resolver by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/3988
* Add extra (unsupported) container build arch's by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4524
* Pass in collection ids to notifier when sharing cipher. by @kristof-mattei in https://github.com/dani-garcia/vaultwarden/pull/4517
* improve access to collections via groups by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4441
* fix emergency access invites by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4337
* Some fixes for the new mobile apps by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4526
* Update Rust, crates and web-vault by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4558
* Improve Commentary Aesthetics by @rich-purnell in https://github.com/dani-garcia/vaultwarden/pull/4549
* Optimize Dockerfiles by @dfunkt in https://github.com/dani-garcia/vaultwarden/pull/4532
* also delete organization_api_key when deleting organizations by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4557
* Fix public api for domains with path prefix by @FDHoho007 in https://github.com/dani-garcia/vaultwarden/pull/4500
* Update crates by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4587
* Fix web-vault version in Docker(files/Settings) by @dfunkt in https://github.com/dani-garcia/vaultwarden/pull/4575
* Update Alpine to version 3.20 by @dfunkt in https://github.com/dani-garcia/vaultwarden/pull/4583
* differentiate external groups by organization id by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4586
* Remove old knowndevice route by @Timshel in https://github.com/dani-garcia/vaultwarden/pull/4578
* Update admin interface dependencies by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4581
* Update rust and remove unused header values by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4645
* Update crates, web-vault and GHA by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4648
* Fix some nightly build errors by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4657
* Fix some more nightly errors and remove lint that will become an error by default by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4661
* Change API and structs to camelCase by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4386
* Fix cipher creation on new android app by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4670
* Remove mimalloc workaround by @dfunkt in https://github.com/dani-garcia/vaultwarden/pull/4606
* Change some missing PascalCase keys by @dani-garcia in https://github.com/dani-garcia/vaultwarden/pull/4671
* Fix collections and native app issue by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4685
* Fix duplicate folder creations during import by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4702
* Remove duplicate registry step by @dfunkt in https://github.com/dani-garcia/vaultwarden/pull/4703
* add group support for Cipher::get_collections() by @stefan0xC in https://github.com/dani-garcia/vaultwarden/pull/4592
* Switch registry cache compression algorithm to zstd by @dfunkt in https://github.com/dani-garcia/vaultwarden/pull/4704
* Update crates and web-vault by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4714
* Some fixes for emergency access by @BlackDex in https://github.com/dani-garcia/vaultwarden/pull/4715

## New Contributors
* @one230six made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4422
* @KrappRamiro made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4408
* @testwill made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4440
* @kristof-mattei made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4517
* @rich-purnell made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4549
* @dfunkt made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4532
* @FDHoho007 made their first contribution in https://github.com/dani-garcia/vaultwarden/pull/4500

**Full Changelog**: https://github.com/dani-garcia/vaultwarden/compare/1.30.5...1.31.0

<hr /><em>This discussion was created from the release <a href='https://github.com/dani-garcia/vaultwarden/releases/tag/1.31.0'>1.31.0</a>.</em>

**Comment by Tealk** [comment]
Where can I find these native mobile app?

**Reply by stefan0xC** [reply]
According to https://bitwarden.com/blog/native-mobile-apps/ you have to join the respective beta group to get access to the app.

**Reply by Tealk** [reply]
Probably not yet available
 
![Screenshot_20240709-080649~2.png](https://github.com/dani-garcia/vaultwarden/assets/12276250/b766b8bb-9cdb-4552-9125-ef722729a6d2)

I already had yesterday, so I thought there was another method

**Reply by BlackDex** [reply]
You can also check the repo's build workflow for the latest build there. Which might work, or not of course.

https://github.com/bitwarden/android/actions/workflows/build.yml
https://github.com/bitwarden/ios/actions/workflows/build.yml

**Reply by dfunkt** [reply]
@Tealk Join the Google group here -> https://groups.google.com/g/bitwarden-beta-testing and click on one of those 2 links in the post.

**Reply by Tealk** [reply]
thanks @dfunkt that worked

---

the login also worked, I find the new design appealing in principle. Unfortunately, the header now takes up a lot of space and the fact that the data of an entry is displayed in text fields looks a bit old-fashioned.

**Comment by sbellon** [comment]
Sorry for being a bit slow here, but what exactly does `This also is an issue when using the previous version of Vaultwarden. You have to update push.bitwarden.eu to api.bitwarden.eu.` mean in the context of a self-hosted Vaultwarden instance?

**Reply by BlackDex** [reply]
Vaultwarden supports sending Push notifications to your mobile, but that goes via the Bitwarden servers as they are needed for that functionality.

**Reply by sbellon** [reply]
But *where* would I have to update that?

**Reply by BlackDex** [reply]
In your settings.
https://github.com/dani-garcia/vaultwarden/blob/d04b94b77d95a9968f61f71969f65aaaa8a496d6/.env.template#L102

**Reply by l4rm4nd** [reply]
I just edited the `config.json` located in the Docker volume and added/adjusted like so:

````
  "push_installation_id": "<MY-ID>",
  "push_installation_key": "<MY-KEY>",
  "push_relay_uri": "https://api.bitwarden.eu",
  "push_identity_uri": "https://identity.bitwarden.eu",
````
Works flawlessly.

**Comment by youcancallmetreris** [comment]
Thank you for the update, works flawlessly here (docker on Synology)

**Comment by ilopmar** [comment]
I can't get either push notifications nor websockets working on my Android phone.

* Websockets: I've tested it works with the desktop application and the web vault. I see on the logs that both applications subscribe to the WS notifications, but the Android app doesn't.
* Push notifications. I've deleted and reinstalled the application to generate the new tokens. I've configured the EU servers and requested the tokens on the Bitwarden app. When I login for the first time with the phone I see that the client is registered. Then, when I login on the web vault and send the "login with device" I don't get the push notification. 
Question: Do I need to have an account on SaaS Bitwarden.com for this to work? Or just putting there my email (the same I use in Vaultwarden) will work?

Thanks!

**Reply by f0ff886f** [reply]
Are you using the beta app or the normal app? I found the normal app on iOS works fine (with the documentation updates applied), but the beta app doesn't work at all for notifications / push changes.

I just used the documentatin to request the installation id with an email address, I have no Bitwarden accounts on their SaaS.

**Reply by ilopmar** [reply]
I tried with both apps, first with the normal app and then with the beta. I don't remember which one was the latest one I used to reset the token and ask for a new one automatically. I'll give it another try to the normal app.

**Reply by Inervo** [reply]
Regarding push notifications: the beta app doesn't seems to works (EU serveur).
It works fine with the normal app, but not with the beta. I've tried to force a "login with device", and in the beta app (no notifications), in "pending login requests", I have the following error: "We were unable to process your request. Please try again or contact us"
